RGB v1/Dio.c

```
@@ -0,0 +1,328 @@
+			{
+				CLEAR_BIT(GPIO_PORTF_GPIODATA, ConfigList.channels[ChannelId].pinId);
+				return DIO_LEVEL_LOW;
+			}
+			else
+			{
+				SET_BIT(GPIO_PORTF_GPIODATA, ConfigList.channels[ChannelId].pinId);
+				return DIO_LEVEL_HIGH;
+			}
+			break;	/* Unreachable */
+	}
+}
+
+/*******************************************************************************
+* Service Name: DioReadPort
+* Service ID[hex]: 0x00
+* Sync/Async: Synchronous
+* Reentrancy: Reentrant
+* Parameters (in): PortId - enuDioPort_t
+* Parameters (inout): None
+* Parameters (out): None
+* Return value: DioPortLevel_t
+* Description: Function for DIO read Port API
+*******************************************************************************/
+DioPortLevel_t DioReadPort(enuDioPort_t PortId)
+{	
+	switch (ConfigList.channels[PortId].portId)
+	{
+		case PORTA:
+			return (DioPortLevel_t)GPIO_PORTA_GPIODATA;
+			break;	/* Unreachable */
+			
+		case PORTB:
+			return (DioPortLevel_t)GPIO_PORTB_GPIODATA;
+			break;	/* Unreachable */
+			
+		case PORTC:
+			return (DioPortLevel_t)GPIO_PORTC_GPIODATA;
+			break;	/* Unreachable */
+			
+		case PORTD:
+			return (DioPortLevel_t)GPIO_PORTD_GPIODATA;
+			break;	/* Unreachable */
+			
+		case PORTE:
+			return (DioPortLevel_t)GPIO_PORTE_GPIODATA;
+			break;	/* Unreachable */
+			
+		case PORTF:
+			return (DioPortLevel_t)GPIO_PORTF_GPIODATA;
+			break;	/* Unreachable */
+	}
+}
+
+/*******************************************************************************
+* Service Name: DioWritePort
+* Service ID[hex]: 0x00
+* Sync/Async: Synchronous
+* Reentrancy: Reentrant
+* Parameters (in): PortId - enuDioPort_t, Level - DioPortLevel_t
+* Parameters (inout): None
+* Parameters (out): None
+* Return value: None
+* Description: Function for DIO write Port API
+*******************************************************************************/
+void DioWritePort(enuDioPort_t PortId, DioPortLevel_t Level)
```

All APIs should return an error state